mlcomp/parallelm/mlpiper/main.py

```diff
@@ -159,9 +172,27 @@ def main(bin_dir=None):
             ml_piper.run_deployment()
 
     elif options.subparser_name in ("run-deployment"):
-        ml_piper = MLPiper(options).deployment_dir(options.deployment_dir).skip_mlpiper_deps_install(True)
+        ml_piper = MLPiper(options) \
+            .deployment_dir(options.deployment_dir) \
+            .skip_mlpiper_deps_install(True) \
+            .test_mode(options.test_mode)
         ml_piper.run_deployment()
 
+    elif options.subparser_name in ("deps"):
+        tmp_dir = tempfile.mkdtemp()
+
+        ml_piper = MLPiper(options) \
+            .comp_repo(options.comp_root) \
+            .deployment_dir(tmp_dir) \
+            .bin_dir(bin_dir) \
+            .pipeline(options.pipeline if options.pipeline else options.file) \
+            .use_color(not options.no_color) \
+            .force(True)
+
+        ml_piper.deps(options.lang)
+        if os.path.exists(tmp_dir) and os.path.isdir(tmp_dir):
```

Comment:
1. `os.path.exists` is redundant
2. You may use `try` & `finally` to ensure tmp dir removal